Sources/SwiftParser/Directives.swift

```diff
@@ -106,29 +109,58 @@ extension Parser {
       while let poundIfHandle = firstIteration ? self.canRecoverTo(.poundIfKeyword) : self.canRecoverTo(anyIn: IfConfigContinuationClauseStartKeyword.self)?.handle,
         loopProgress.evaluate(self.currentToken)
       {
-        var (unexpectedBeforePoundIf, poundIf) = self.eat(poundIfHandle)
-        firstIteration = false
-        // Parse the condition.
+        var unexpectedBeforePoundIf: RawUnexpectedNodesSyntax?
+        var poundIf: Parser.Token
         let condition: RawExprSyntax?
-        switch poundIf.tokenKind {
-        case .poundIfKeyword, .poundElseifKeyword:
+
+        var atElifTypo: Bool {
+          guard self.at(.pound) else {
+            return false
+          }
+
+          var lookahead = self.lookahead()
+          lookahead.consumeAnyToken()
+
+          if lookahead.at(.identifier), lookahead.currentToken.tokenText == "elif" {
+            lookahead.consumeAnyToken()
+          } else {
+            return false
+          }
+
+          return lookahead.at(.identifier)
+        }
+
+        if atElifTypo {
+          (unexpectedBeforePoundIf, poundIf) = self.eat(poundIfHandle)
+          guard let identifier = self.consume(if: TokenSpec(.identifier, allowAtStartOfLine: false)) else {
```

Comment:
Hey @ahoppen, I've added the following tests and they are passing, but I think they shouldn't:
```swift
  func testIfConfigExpr34() {
    AssertParse(
      """
      #if MY_FLAG
      #
      elif
      #endif
      """
    )
  }
  
  func testIfConfigExpr35() {
    AssertParse(
      """
      #if MY_FLAG
      # elif
      #endif
      """
    )
  }
```

Reply:
Could you share what kind of syntax trees these produce? Either with `swift-parser-cli print-tree /path/to/file/containing/snippet.swift` or by setting a breakpoint in `AssertParse` and running `e print(tree.recursiveDescription)`

Reply:
Sure. For the code in `testIfConfigExpr34`, it prints the following tree:
```swift
SourceFileSyntax children=2
  0: CodeBlockItemListSyntax children=1
    0: CodeBlockItemSyntax children=1
      0: IfConfigDeclSyntax children=2
        0: IfConfigClauseListSyntax children=1
          0: IfConfigClauseSyntax children=3
            0: poundIfKeyword
            1: IdentifierExprSyntax children=1
              0: identifier("MY_FLAG")
            2: CodeBlockItemListSyntax children=1
              0: CodeBlockItemSyntax children=1
                0: MacroExpansionExprSyntax children=3
                  0: pound
                  1: identifier("elif")
                  2: TupleExprElementListSyntax
        1: poundEndifKeyword
  1: eof
```

And for the code in `testIfConfigExpr35` it prints this:
```swift
SourceFileSyntax children=2
  0: CodeBlockItemListSyntax children=1
    0: CodeBlockItemSyntax children=1
      0: IfConfigDeclSyntax children=2
        0: IfConfigClauseListSyntax children=1
          0: IfConfigClauseSyntax children=3
            0: poundIfKeyword
            1: IdentifierExprSyntax children=1
              0: identifier("MY_FLAG")
            2: CodeBlockItemListSyntax children=1
              0: CodeBlockItemSyntax children=1
                0: MacroExpansionExprSyntax children=3
                  0: pound
                  1: identifier("elif")
                  2: TupleExprElementListSyntax
        1: poundEndifKeyword
  1: eof
```

They look like the same output.

Reply:
OK, nice. This looks correct as far as your PR is concerned since we are parsing `#elif` as a macro expansion. I filed https://github.com/apple/swift-syntax/issues/1395 to keep track of the issue. It’s similar to what you are doing here so  if you want to make another contribution, I think it would be a good candidate for you.

Reply:
Could you add a FIXME to these test cases mentioning that they should generate a diagnostic and link to #1395?

Reply:
Nice @ahoppen, thank you. I'll start working on it.

Reply:
Awesome, thanks!